Description: export CC pointing to the kernel's compiler
 export the CC variable from .kernelvariables in the kernel source to allow
 module build systems use the kernel's compiler also outside of Kbuild
Author: Andreas Beckmann <anbe@debian.org>

--- dkms-2.8.4.orig/dkms
+++ dkms-2.8.4/dkms
@@ -1157,6 +1157,12 @@ prepare_kernel()
     esac
     }
 
+    if [ -f "$kernel_source_dir/.kernelvariables" ]; then
+        export CC=$(echo -e "show-%:\n\t@echo \$(\$*)\ninclude $kernel_source_dir/.kernelvariables" | make -f - show-CC)
+    else
+        unset CC
+    fi
+
     [[ $no_prepare_kernel ]] && return
 
     if [[ (! ( $(VER $1) < $(VER 2.6.5) ) || -d /etc/SuSEconfig) && \
